suspicion.go

```diff
@@ -0,0 +1,89 @@
+package memberlist
+
+import (
+	"math"
+	"sync/atomic"
+	"time"
+)
+
+// suspicion manages the suspect timer for a node and provides an interface
+// to accelerate the timeout as we get more independent confirmations that
+// a node is suspect.
+type suspicion struct {
+	// k is the number of independent confirmations we'd like to see in
+	// order to drive the timer to its minimum value.
+	k int32
+
+	// min is the minimum timer value in seconds.
+	min float64
```

Comment:
All the math is on them as floats but `time.Duration` is probably more readable / safer for later maintenance so I'll change it.